### DISCUSSION TITLE
[ERROR] Incorrect ics/ical URL

### DISCUSSION THREAD

**Discussion by FootlockerX** [derekantrican/GAS-ICS-Sync#387] [Q&A] [answered]
Hi, first of all, amazing project and it has been working flawlessly for about 1½ years for me.

I recently started to encounter som errors with the script:
[ERROR] Incorrect ics/ical URL: https://outlook.office365.com/owa/calendar/secret/calendar.ics

I have replaced the original variable with secret in the above URL in order not to share my work calendar publicly
When i manually input the calendar.ics url in a webbrowser it downloads an calendar.ics from microsoft with all the events in it.
the URL does have a long response time for 15-25 secs, but it is working.

my problem is why is the script throwing incorrect URL, when the calendar.ics is suppositly fine?

when looking at the helpers.gs code line 74-100 listed below

"
callWithBackoff(function() {
      var urlResponse = UrlFetchApp.fetch(url, { 'validateHttpsCertificates' : false, 'muteHttpExceptions' : true });
      if (urlResponse.getResponseCode() == 200){
        var urlContent = RegExp("(BEGIN:VCALENDAR.*?END:VCALENDAR)", "s").exec(urlResponse.getContentText());
        if(urlContent == null){
          Logger.log("[ERROR] Incorrect ics/ical URL: " + url);
          return;
"

if I read it correctly, it should only throw the incorrect ics/ical URL if the http returns 200 and the url content is null.

dont know how to proceed from here, is there anyone with more knowledge on this subject and willing to help me?

**Comment by jonas0b1011001** [comment, marked as answer]
Is this by chance related to #386 ?

**Reply by FootlockerX** [reply]
Hi Jonas
100%, and the fix posted is working. :)

thanks for your help
